### DISCUSSION TITLE
Local testing cloudtrail events against a policy

### DISCUSSION THREAD

**Discussion by nitrocode** [cloud-custodian/cloud-custodian#7382] [General]
Hi. I'd like to put in a cloudtrail event into a json file and use it as an input to custodian to see how it will run against my policy.

I'd like to do a dryrun test and a non-dryrun.

Is this possible to do locally? If so, how would I do this ?

**Comment by kapilt** [comment]
i'm not sure what that would mean, the cloudtrail event would need to correspond to infrastructure already provisioned, ie the event doesn't exist in isolation, so its unclear what utility that specifying a trail event would have by itself. we're looking at #4912 which would be a basis of policy testing, inclusive of event modes.
